src/app/pages/Project.tsx

```diff
@@ -1,51 +1,98 @@
 import React, {useState, useEffect} from 'react';
 import {
+  fetchProject,
   createDocument,
   deleteDocument,
   fetchDocumentsProject,
+  fetchDocumentsAndMultiPacksProject,
+  createCrossDoc,
+  deleteCrossDoc,
 } from '../lib/api';
 import {Link, useHistory} from 'react-router-dom';
 import DropUpload from '../components/dropUpload';
 import {FileWithPath} from 'react-dropzone';
 
 function Docs() {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [projectInfo, setProjectInfo] = useState<any>();
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const [docs, setDocs] = useState<any[]>([]);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [crossdocs, setCrossDocs] = useState<any[]>();
 
   const history = useHistory();
 
   useEffect(() => {
-    updateDocs().catch(() => {
+    getProjectInfo().catch(() => {
       history.push('/login');
     });
   }, [history]);
 
-  function updateDocs() {
-    const project_id = window.location.pathname.split('/').pop()!;
+  useEffect(() => {
+    updateDocs();
+  }, [projectInfo]);
 
-    return fetchDocumentsProject(project_id).then(docs => {
-      setDocs(docs);
+  function getProjectInfo() {
+    const project_id = window.location.pathname.split('/').pop()!;
+    return fetchProject(project_id).then(info => {
+      setProjectInfo(info);
     });
   }
-
-  function handleAdd(filesToUpload: FileWithPath[]) {
-    const project_id = window.location.pathname.split('/').pop()!;
-
-    filesToUpload.forEach(f => {
-      const reader = new FileReader();
-      reader.readAsText(f);
-      reader.onload = function () {
-        createDocument(f.name, reader.result as string, project_id).then(() => {
-          updateDocs();
+  function updateDocs() {
+    if (projectInfo) {
```

Comment:
if we make `projectInfo` the parameter of `updateDocs` function, then we don't need this clunky if check.

Reply:
does `function updateDocs(projectInfo) {` work?

Reply:
resolved in the newer commit